### PR TITLE
Added vergen and used that to read the GIT_SHA for version string in …

### PR DESCRIPTION
**Pull request by efrisch**
…clap.

## PR Checklist

- [x] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
- [x] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
- [x] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
- [x] I've not introduced any new known security vulnerabilities. I've run `cargo audit`

- [x] I've included a brief description and a good title of the proposed changes and how to test them.
- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
- [x] I've checked that the code is up-to-date with the `pyrsia/main` branch.
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've assigned this Pull Request to "pyrsia/collaborators"

## Description

Fixes pyrsia/pyrsia#490

This PR adds the git version sha hash by using vergen and a custom build.rs to read it at build.

## Screenshots (optional)

Screen Shot 2022-03-22 at 10 10 55 AM

### CODE REVIEW

File: pyrsia_cli/Cargo.toml [hunk 1]
```diff
@@ -24,4 +26,6 @@ path = "src/main.rs"
 directories = "^2.0"
 assay = "0.1.1"
 
-
+[build-dependencies]
+anyhow = "1.0"
+vergen = "7.0.0"
```
Comment (fishseabowl) [inline pyrsia_cli/Cargo.toml:31]: No newline at end of file

**Review by sbtaylor15 — APPROVED** [review]
/lgtm - no other changes needed

**Review by fishseabowl — APPROVED** [review]
It looks fine to me

File: pyrsia_cli/src/cli/parser.rs [hunk 2]
```diff
@@ -14,9 +14,11 @@
    limitations under the License.
 */
 
-use clap::{arg, command, AppSettings, ArgMatches, Command};
+use clap::{arg, command, crate_version, AppSettings, ArgMatches, Command};
+use const_format::formatcp;
 
 pub fn cli_parser() -> ArgMatches {
+    let version_string: &str = formatcp!("{} ({})", crate_version!(), env!("VERGEN_GIT_SHA"));
```
Comment (prince-chrismc) [inline pyrsia_cli/src/cli/parser.rs:21]: I was expecting this to match with the deb packing version?  

https://github.com/pyrsia/pyrsia/blob/0e5ede84384c73ff3fcbb229f7fb99f3d83e5235/.github/workflows/rust.yml#L76-L78

**Review by betarelease — APPROVED** [review]

**Pull request merged**